cmd/kv-tcp/main.go

```diff
@@ -107,14 +159,16 @@ func (c conn) serve(ctx context.Context) {
 				continue
 			}
 
-			cmd, err := c.GetCommand(ctx, n, p1, p2)
+			cmd, err := c.GetCommand(cctx, n, p1, p2)
 			if err != nil {
 				logrus.Warnln(err)
 				fmt.Fprintf(c.nc, "%v\r\n", err)
 				return
 			}
 
-			err = cmd.Execute(ctx)
+			err = cmd.Execute(context.WithValue(cctx, stores.ContextKeyTransactionID, c.txID))
+			srv := ctx.Value(ctxKeyServer).(server)
```

Comment:
At this point, I think the store should just live on the `conn`. All of this indirection to get to it in context is a bit expensive now.